### ISSUE TITLE
RunTimeError is misspelled in the test status files

### ISSUE THREAD

**Issue by kmillikin** [dart-lang/sdk#31374]
The proper camel case spelling of "run-time error" should be RunTimeError and it should be changed in the test status files.

If not, we should then switch to CompiletimeError and MissingCompiletimeError (which is horrible) for consistency.

**Comment by munificent** [comment]
**TL;DR: "RuntimeError" and "CompileTimeError" are each consistent with *current usage* even though not consistent with *each other*.**

I have [sunk more of my life](https://github.com/munificent/craftinginterpreters/blob/master/note/style%20guide.txt#L46-L66) into hyphenation and concatenation rules that probably anyone should, mainly because I need to make sure I apply a correct, consistent style in my books.

The short answer is that there are no context-free rules for which word pairs are separate words ("real estate"), hyphenated ("three-year-old son"), or single words ("website"). The general rules are:

* Over time, a new compound word begins as separate words ("web site"), becomes hyphenated ("web-site"), and eventually coalesces into a single word ("website") if it is frequently used enough. So the answer for how a compound word is punctuated is mostly a matter of history and popularity. The "correct" answer for any given word usually requires consulting word lists in a style guide like the Chicago Manual of Style.

* Compound words that have not become a single word ("keyboard") are typically hyphenated when used as an adjective preceding a noun ("up-to-date schedule") but not when used as a standalone noun phrase ("schedule is up to date"). This is not *always* true, but it usually is. It's "post office box", not "post-office box" even though "post office" is a compound word.

In this specific case, we're in a weird spot because "run time" is in a transition period. [Wikipedia lists both "run time" and "runtime"](https://en.wikipedia.org/wiki/Run_time_(program_lifecycle_phase\)) when referring to the execution phase as a noun. When referring to the part of the system, [they use "runtime" or "run-time"](https://en.wikipedia.org/wiki/Runtime_system). [The examples related to computing in the OED use "run-time" for both the noun and adjective forms](https://en.oxforddictionaries.com/definition/runtime).

Since our readers are in a computing context, I think it's best to follow the norms of other tech sources, since the wider English community tends to lag that. We're using "runtime" as an adjective "runtime error", so looking for that form, I see:

* Our own core library has a member ["runtimeType"](https://api.dartlang.org/stable/1.24.2/dart-core/Object/runtimeType.html). (lowercase "t"). ("Lowercase" is itself a compound word that [recently transitioned from an open to closed form](https://books.google.com/ngrams/graph?content=lower+case%2Clowercase&year_start=1800&year_end=2000&corpus=15&smoothing=3&share=&direct_url=t1%3B%2Clower%20case%3B%2Cc0%3B.t1%3B%2Clowercase%3B%2Cc0).)

* Python calls the exception ["RuntimeError"](https://docs.python.org/3/library/exceptions.html).

* Ruby also uses ["RuntimeError"](https://ruby-doc.org/core-2.2.0/RuntimeError.html).

* IE reports it as a ["Runtime Error"](https://support.microsoft.com/en-us/help/822521/fix-runtime-errors-in-internet-explorer).

Given that, I think it should be "runtime", no hyphen or space, when used as an adjective. When CamelCased, the "t" stays lowercase.

"Compile time error" is different. If you Google, most people tend to use "compile error" and omit "time" entirely. I see [2,380,000](https://www.google.com/search?biw=1429&bih=803&ei=VScoWrexDdXqjwOGuZbgDA&q=compile+time+error&oq=compile+time+error&gs_l=psy-ab.3..0l10.694.1099.0.1204.5.5.0.0.0.0.106.380.4j1.5.0....0...1c.1.64.psy-ab..0.5.379...0i67k1j0i20i264k1j0i7i30k1j0i13k1.0.0FYVqdZc8mg) results for "compile time error", [2,430,000] results for "compile-time error", and only [16,700](https://www.google.com/search?biw=1429&bih=803&ei=VycoWve1Kpi0jwOX2LXwDA&q=compiletime+error&oq=compiletime+error&gs_l=psy-ab.3..35i39k1j0i7i30k1l9.12012.12012.0.12202.1.1.0.0.0.0.52.52.1.1.0....0...1c.1.64.psy-ab..0.1.51....0.oAS4BagM2IU) results for "compiletime error".

"Compile time" as a compound word is much less common than "run time" (unsurprising since many more people run programs than compile them), so it doesn't appear to have gone through the coalescing process that "runtime" has yet. Maybe it will at some point, but the idiomatic style today is still open form.

If the discrepancy really bothers you, one way to dodge it is to use the more common term "compile error" and eliminate "time" entirely. It's also shorter. "Static error" would work too.

Now, when are we going to talk about "timeout"? :)

**Comment by kmillikin** [comment]
I do think self-consistency is important and that parallel things should use parallel constructions.  It helps the reader because they don't have to wonder if you're making an important distinction or not.  And it helps the writer on a team of writers because they don't have to guess what to write.

Here it's not much effort and I don't think we'd be too far away from common usage if we just adopted the simple rules: (1) "run time" and "compile time" are noun phrases describing times, because they are parallel and "compiletime" is ugly, (2) the adjective forms are "run-time" and "compile-time" because they are parallel, they are a single adjective not two, and "compiletime" is ugly.

Then we can ask how hyphenated words are camel-cased.  I don't really know what the rule is, but it should be consistent and, again, "compiletime" is still ugly.

We do use "runtime" in Dart, but we use it as jargon describing the "run-time system" like the VM or the dart2js JS runtime.  I think that's fine, but here it's parallel to "compiler".  If we use the same nouns "runtime" for the time and "runtime" for a Dart run-time system, then we lose the ability to distinguish the adjective forms "runtime" and "runtime" which I would actually miss.  (The noun forms are easy because "runtime" as in "run-time system" normally has an article and "runtime" as in "run time" normally does not.)

(I also think `runtimeType` was a mistake because that's a type used in the run-time system, but nobody asked me.)

**Comment by kmillikin** [comment]
Or else we get way out ahead of the curve and go with `runtimeerror`, `compiletimeerror`, and `missingCompiletimeerror`.

**Comment by munificent** [comment]
>  I don't think we'd be too far away from current usage

* [The O'Reilly style guide](http://chimera.labs.oreilly.com/books/1230000000969/ch04.html) for their books uses "runtime" for both noun and adjective use, "compile time" for a noun, and "compile-time" for adjective.

* [The Google Developer Docs style guide](https://developers.google.com/style/word-list), which is [the style guide we use for our own docs](https://github.com/dart-lang/site-www/blob/master/CONTRIBUTING.md) does the same.

I think those are about as authoritative as we could hope to be.
